.github/workflows/pull_request.yml

```diff
@@ -71,18 +79,20 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Installation
-        run: |
-          sudo apt-get update -qq
-          sudo apt-get install -y zfsutils-linux
-          truncate -s 100G /tmp/disk.img
-          sudo zpool create zfspv-pool `sudo losetup -f /tmp/disk.img --show`
-
       - name: Set up Go 1.19
         uses: actions/setup-go@v4
         with:
           go-version: 1.19.9
           cache: false
+        
+      - name: Build images locally
+        run: make zfs-driver-image || exit 1;
+
+      - name: Setup zfs pools
+        run: |
+          sudo apt-get install -y zfsutils-linux
+          truncate -s 100G /tmp/disk.img
+          sudo zpool create zfspv-pool `sudo losetup -f /tmp/disk.img --show`
```

Comment:
Should this be behind the nix-shell?

Reply:
Why should this be behind nix shell?

Reply:
`apt-get` is tied to debian/ubuntu. And we are adding the nix-shell to pull dependencies. Could it sit beside semver and yq, and get pulled in together?

Reply:
But the tests and image build is happening outside nix-shell, unless we want to put everything behind nix-shell, which I would not prefer as that would need more changes.

Reply:
Fair enough. Makes sense.